py/desiutil/test/test_bitmask.py

```diff
@@ -181,3 +180,28 @@ def test_print(self):
         for i, name in enumerate(self.ccdmask.names()):
             self.assertEqual(str(self.ccdmask[name]), bit_str[i])
             self.assertEqual(repr(self.ccdmask[name]), bit_repr[i])
+
+    def test_numpy_compat(self):
+        """Test compatibility with Numpy 1 and Numpy 2.
+        """
+        signed_array_16 = np.zeros((10,), dtype=np.int16)
+        signed_array_32 = np.zeros((10,), dtype=np.int32)
+        signed_array_64 = np.zeros((10,), dtype=np.int64)
+        unsigned_array_16 = np.zeros((10,), dtype=np.uint16)
+        unsigned_array_32 = np.zeros((10,), dtype=np.uint32)
+        unsigned_array_64 = np.zeros((10,), dtype=np.uint64)
+        for array in (signed_array_16, signed_array_32, signed_array_64,
+                      unsigned_array_16, unsigned_array_32, unsigned_array_64):
+            new_array = array.copy()
+            new_array |= self.ccdmask.COSMIC
+            new_array |= self.ccdmask.HOT
+            self.assertTrue((new_array == (array + 18)).all())
+            self.assertTrue((np.bitwise_and(new_array, self.ccdmask.HOT) != 0).all())
+            self.assertTrue((np.bitwise_and(self.ccdmask.HOT, new_array) != 0).all())
+            out_array = array.copy()
+            np.bitwise_xor(new_array, self.ccdmask.BAD, out=out_array)
+            self.assertTrue((out_array == (array + 19)).all())
```

Comment:
Let's also add tests that combinations of bits also work prior to the operation with the array (they do work, but this is tricky enough code that I want to check corner cases).  e.g.
```
new_array |= self.ccdmask.mask('DEAD|SATURATED')
new_array |= (self.ccdmask.DEAD | self.ccdmask.SATURATED)
```
Let's also explicitely check the syntax like
```
result = (new_array & self.ccdmask.COSMIC)
```
Under the hood I think that is doing the same thing as `result = np.bitwise_and(new_array, self.ccdmask.COSMIC)` but since we're modifying the behavior of low-level operators, let's make sure.

Reply:
Agreed.